src/style.rs

```diff
@@ -129,6 +193,12 @@ impl ProgressStyle {
         self
     }
 
+    /// Adds a custom key parser that references a `&ProgressState` to the template.
+    pub fn key(mut self, key: &'static str, f: Format) -> ProgressStyle {
```

Comment:
Let's call this `with_key()` (and please remove the period from the docstring).

Reply:
Done.